conduit/data/datasets/vision/camelyon17.py

```diff
@@ -0,0 +1,170 @@
+    _VAL_CENTER: ClassVar[int] = 1
+
+    @parsable
+    def __init__(
+        self,
+        root: Union[str, Path],
+        *,
+        download: bool = True,
+        transform: Optional[ImageTform] = None,
+        split: Optional[Union[Camelyon17Split, str]] = None,
+        split_scheme: Union[Camelyon17SplitScheme, str] = Camelyon17SplitScheme.official,
+        superclass: Union[Camelyon17Attr, str] = Camelyon17Attr.tumor,
+        subclass: Union[Camelyon17Attr, str] = Camelyon17Attr.center,
+    ) -> None:
+
+        self.superclass = str_to_enum(str_=superclass, enum=Camelyon17Attr)
+        self.subclass = str_to_enum(str_=subclass, enum=Camelyon17Attr)
+
+        self.split = (
+            str_to_enum(str_=split, enum=Camelyon17Split) if isinstance(split, str) else split
+        )
+        self.split_scheme = (
+            str_to_enum(str_=split_scheme, enum=Camelyon17SplitScheme)
+            if isinstance(split_scheme, str)
+            else split_scheme
+        )
+        self.root = Path(root)
+        self._base_dir = self.root / "camelyon17_v1.0"
+        self.download = download
+        if self.download:
+            download_from_url(
+                file_info=self._FILE_INFO,
+                root=self.root,
+                logger=self.logger,
+                remove_finished=True,
+            )
+        else:
+            raise FileNotFoundError(
+                f"Data not found at location {self._base_dir.resolve()}. Have you downloaded it?"
+            )
+
+        # Read in metadata
+        # Note: metadata is one-indexed.
+        self.metadata = pd.read_csv(
+            self._base_dir / 'metadata.csv', index_col=0, dtype={"patient": "str"}
+        )
+        if self.split_scheme is Camelyon17SplitScheme.mixed_to_test:
+            # For the mixed-to-test setting,
+            # we move slide 23 (corresponding to patient 042, node 3 in the original dataset)
+            # from the test set to the training set
+            slide_mask = self.metadata["slide"] == 23
+            self.metadata.loc[slide_mask, "split"] = Camelyon17Split.train.value
+        # Use an official split of the data, if 'split' is specified, else just use all
+        # of the data
+        val_center_mask = self.metadata["center"] == self._VAL_CENTER
+        test_center_mask = self.metadata["center"] == self._TEST_CENTER
+        self.metadata.loc[val_center_mask, "split"] = Camelyon17Split.val.value
+        self.metadata.loc[test_center_mask, "split"] = Camelyon17Split.test.value
+
+        if self.split is not None:
+            split_indices = self.metadata["split"] == self.split.value
+            self.metadata = cast(pd.DataFrame, self.metadata[split_indices])
+
+        # Construct filepaths from metadata
+        def build_fp(row: pd.DataFrame) -> str:
+            return "patches/patient_{0}_node_{1}/patch_patient_{0}_node_{1}_x_{2}_y_{3}.png".format(
+                *row
+            )
```

Comment:
I don't love this, but I can offer no better solution

Reply:
Believe me, this is far nice than the original implementation (derived from WILDS) which made use of ``iiter_tuples``